Common/src/adt_structure.cpp

```diff
@@ -849,14 +849,18 @@ void CADTElemClass::DetermineNearestElement(const su2double *coor,
           coorBBMin = BBoxCoor.data() + nDimADT*kk;
           coorBBMax = coorBBMin + nDim;
 
-          su2double posDist2 = 0.0;
+          su2double posDist2 = 0.0, ds;
+          ds = min(0.0, coor[0]-coorBBMin[0]) + max(0.0, coor[0]-coorBBMax[0]); posDist2 += ds*ds;
+          ds = min(0.0, coor[1]-coorBBMin[1]) + max(0.0, coor[1]-coorBBMax[1]); posDist2 += ds*ds;
+          if(nDim==3) {
+          ds = min(0.0, coor[2]-coorBBMin[2]) + max(0.0, coor[2]-coorBBMax[2]); posDist2 += ds*ds;}
+          /*--- The above is equivalent to this:
           for(unsigned short k=0; k<nDim; ++k) {
             su2double ds = 0.0;
             if(     coor[k] < coorBBMin[k]) ds = coor[k] - coorBBMin[k];
             else if(coor[k] > coorBBMax[k]) ds = coor[k] - coorBBMax[k];
-
             posDist2 += ds*ds;
-          }
+          }, but a bit faster. ---*/
```

Comment:
Now the single largest time consuming preprocessing step is wall distance computation, so here are some micro optimizations of a very hot function.